### PR TITLE
Add the upcoming workspace functionality to the assistant API client

### PR DESCRIPTION
**Pull request by bzvestey**
This adds all the workspace functionality that the Refactor assistant should need.

### CODE REVIEW

File: src/AssistantAPIClient.js [hunk 1]
```diff
@@ -97,7 +97,40 @@ class AssistantAPIClient {
...
+
+  /**
+   * Returns a list of user accessible Workspaces.  By default it will just be
+   * the user owned Workspaces, but can be configured to also return all the
+   * public workspaces.
+   *
+   * @param {boolean} includePublic When true the returned list includes public Workspaces.
+   * @return {Array<Workspace>} The list of Workspaces.
+   */
+  userAccessibleWorkspaces(includePublic = false) {
```
Comment (LoganGore) [inline src/AssistantAPIClient.js]: This should follow similar naming conventions like we have throughout the API, like `getUserAccessibleWorkspaces`.

Comment (bzvestey) [reply src/AssistantAPIClient.js]: Ok, I was trying to keep it from too long, but I can update it.

Comment (LoganGore) [reply src/AssistantAPIClient.js]: Well, maybe you could drop the 'user' then. Isn't 'user' accessibly implied with the fact that they're using the api client?

Comment (bzvestey) [reply src/AssistantAPIClient.js]: I feel like having the user in there makes it flow better.

**Pull request merged**